addons/adyen_marketpay/models/adyen_account.py

```diff
@@ -0,0 +1,640 @@
+        account. After logging in on odoo.com the user will be redirected to his DB with a token in
+        the URL. This token is then needed to create the Adyen account.
+        '''
+        if self.env.company.adyen_account_id:
+            raise UserError(_('You can only create a single Adyen account per company.'))
+        return_url = url_join(self.env['ir.config_parameter'].sudo().get_param('web.base.url'), 'adyen_marketpay/create_account')
+        onboarding_url = self.env['ir.config_parameter'].sudo().get_param('adyen_marketpay.onboarding_url')
+        return {
+            'type': 'ir.actions.act_url',
+            'url': url_join(onboarding_url, 'get_creation_token?return_url=%s' % return_url),
+        }
+
+    def action_show_transactions(self):
+        return {
+            'name': _('Transactions'),
+            'view_mode': 'tree,form',
+            'domain': [('adyen_account_id', '=', self.id)],
+            'res_model': 'adyen.transaction',
+            'type': 'ir.actions.act_window',
+        }
+
+    def _upload_document(self, document_type, content, filename):
+        self._adyen_rpc('upload_document', {
+            'documentDetail': {
+                'accountHolderCode': self.account_holder_code,
+                'documentType': document_type,
+                'filename': filename,
+            },
+            'documentContent': content,
+        })
+
+    def _format_data(self):
+        data = {
+            'accountHolderCode': self.account_holder_code,
+            'accountHolderDetails': {
+                'address': {
+                    'country': self.country_id.code,
+                    'stateOrProvince': self.state_id.code or None,
+                    'city': self.city,
+                    'postalCode': self.zip,
+                    'street': self.street,
+                    'houseNumberOrName': self.house_number_or_name,
+                },
+                'email': self.email,
+                'fullPhoneNumber': self.phone_number,
+            },
+            'legalEntity': 'Business' if self.is_business else 'Individual',
+        }
+
+        if self.is_business:
+            data['accountHolderDetails']['businessDetails'] = {
+                'legalBusinessName': self.legal_business_name,
+                'doingBusinessAs': self.doing_business_as,
+                'registrationNumber': self.registration_number,
+            }
+        else:
+            data['accountHolderDetails']['individualDetails'] = {
+                'name': {
+                    'firstName': self.first_name,
+                    'lastName': self.last_name,
+                    'gender': 'UNKNOWN',
+                },
+                'personalData': {
+                    'dateOfBirth': str(self.date_of_birth),
+                }
+            }
+
+            # documentData cannot be present if not set
+            if self.document_number:
```

Comment:
There is a key error on "businessDetails", you are under indiviualDetailes, shouldn't it be in the is_business? or it should probably be set on individualDetails

Reply:
but there are not supposed to have shareholders in indiviual i think

Reply:
Indeed there's something wrong here, I copied it from adyen.shareholder model and probably forgot to change the key